src/Trace/Trace.php

```diff
@@ -0,0 +1,171 @@
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+use Google\Cloud\Core\ValidateTrait;
+
+/**
+ * This plain PHP class represents a Trace resource. Traces belong to a
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ */
+class Trace
+{
+    use ValidateTrait;
+
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param string $projectId The id of the project this trace belongs to.
+     * @param array $options [optional] {
+     *      Configuration options.
+     *
+     *      @type string $traceId The trace id for this trace. 128-bit numeric
+     *            formatted as a 32-byte hex string. If not provided, one will be generated
+     *            automatically for you.
+     *      @type array $spans List of span data to load.
+     * }
+     */
+    public function __construct($projectId, array $options = [])
+    {
+        $this->projectId = $projectId;
+
+        if (array_key_exists('traceId', $options)) {
+            $this->traceId = $options['traceId'];
+        }
+        $this->traceId = $this->traceId ?: $this->generateTraceId();
+
+        if (array_key_exists('spans', $options)) {
+            foreach ($options['spans'] as $span) {
+                array_push($this->spans, new TraceSpan($span));
+            }
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function traceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Retrieves the trace's projectId.
+     *
+     * @return string
+     */
+    public function projectId()
+    {
+        return $this->projectId;
+    }
+
+    /**
+     * Set the trace's projectId
+     * @param string $projectId
+     */
+    public function setProjectId($projectId)
+    {
+        $this->projectId = $projectId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        return [
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId,
+            'spans' => array_map(function ($span) {
+                return $span->info();
+            }, $this->spans)
+        ];
+    }
+
+    /**
+     * Retrieves the spans for this trace.
+     *
+     * @return array
+     */
+    public function spans()
+    {
+        return $this->spans;
+    }
+
+    /**
+     * Set the spans for this trace.
+     *
+     * @param TraceSpan[] $spans
+     */
+    public function setSpans(array $spans)
```

Comment:
Just an additional helper, something along the lines of this: https://github.com/GoogleCloudPlatform/google-cloud-php/blob/master/src/Vision/VisionClient.php#L140

It could be helpful in cases where someone may want to construct spans and have low level control outside of the context manager.